service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcRequestBuilder.java

```diff
@@ -95,7 +97,19 @@ private ParametersParameterComponent buildParamterComponent(String parameterName
         BodyInserter<Object, ReactiveHttpOutputMessage> bodyInserter
             = BodyInserters.fromValue(requestBody);
 
-        return buildRequestWithHeadersAndBody(uri, requestBody, bodyInserter, structuredTaskDefinition);
+        return buildRequestWithHeadersAndBody(uri, requestBody, bodyInserter, structuredTaskDefinition, GPC_STRUCTURED_INTERACTION_ID);
+    }
+
+    public RequestHeadersSpec<?> buildGetDocumentRecordRequest(GetGpcDocumentTaskDefinition documentTaskDefinition) {
+        SslContext sslContext = buildSSLContext();
+        HttpClient httpClient = HttpClient.create().secure(t -> t.sslContext(sslContext));
+        WebClient client = buildWebClient(httpClient);
+
+        WebClient.RequestBodySpec uri = client
```

Comment:
if we separate this to another method accepting url and interactionid we could use it for both structured and document

Reply:
Structured uses POST method and Access uses "GET". There is no much to move to another method.